controllers/openstackmachine_controller.go

```diff
@@ -583,7 +583,28 @@ func (r *OpenStackMachineReconciler) getOrCreateMachineServer(ctx context.Contex
 			}
 			return openStackCluster.Spec.IdentityRef
 		}()
-		machineServerSpec := openStackMachineSpecToOpenStackServerSpec(&openStackMachine.Spec, identityRef, compute.InstanceTags(&openStackMachine.Spec, openStackCluster), failureDomain, userDataRef, getManagedSecurityGroup(openStackCluster, machine), openStackCluster.Status.Network.ID)
+		// If user has provided .spec.ports (non-empty), skip .Status.Network entirely
+		// Otherwise, fall back to openStackCluster.Status.Network.ID
+		// This supports HCP - Hosted Control Plane usage
```

Comment:
Let's write a more generic comment, please.
In general, let's remove "HPC / HCP" from the whole PR. Let's consider it a generic use case where the control plane is not managed by CAPO.